src/App/App.js

```diff
@@ -33,21 +38,52 @@ export default class App extends Component {
     this.setState({
       goods: newArray,
       total: getTotal(newArray),
+      subTotal: getSubTotal(newArray)
+    })
+  }
+
+  onDeleteSelected = () => {
+    const newArray = removeSelected(this.state.goods)
+    this.setState({
+      goods: newArray,
+      total: getTotal(newArray),
+      subTotal: 0
+    })
+  }
+
+  onToggle = (id) => {
+    toggleCheckbox(id, this.state.goods)
+    this.setState({
+      subTotal: getSubTotal(this.state.goods)
+    })
+  }
```

Comment:
toggleCheckbox **shouldn't mutate** original state
new goods array should be set next to subtotal